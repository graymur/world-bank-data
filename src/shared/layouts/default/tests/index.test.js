/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';
import {Layout} from '../index';

test('Renders content in wrapper', () => {
	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<Layout><h1>content</h1></Layout>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Renders error component if error is passed in props', () => {
	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<Layout error={'Error text'}><h1>content</h1></Layout>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

// test('Calls setError function if child component throws error', () => {
// 	const setError = jest.fn();
//
// 	/**
// 	 * React does "console.error" error caught in componentDidCatch.
// 	 * It doesn't affect the test but is very annoying. This is a rude way to mute it.
// 	 * JSDom will do it's own "console.log", didn't find a way to mute it.
// 	 */
// 	const consoleError = global.console.error;
// 	global.console.error = () => {};
//
// 	const ComponentWithError = () => {
// 		throw new Error('Some error');
// 	};
//
// 	const component = renderer.create(
// 		<StaticRouter location="someLocation" context={{}}>
// 			<Layout setError={setError}><ComponentWithError/></Layout>
// 		</StaticRouter>
// 	);
//
// 	expect(setError.mock.calls.length).toBe(1);
// 	expect(setError.mock.calls[0][0].message).toBe('Some error');
//
// 	global.console.error = consoleError;
// });
