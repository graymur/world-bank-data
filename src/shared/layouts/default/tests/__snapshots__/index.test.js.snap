// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders content in wrapper 1`] = `
<div
  className="main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <nav
        className="main-nav"
      >
        <a
          aria-current={false}
          className="main-nav__item"
          href="/"
          onClick={[Function]}
          style={undefined}
        >
          Main page
        </a>
        <a
          aria-current={false}
          className="main-nav__item"
          href="/countries"
          onClick={[Function]}
          style={undefined}
        >
          Countries
        </a>
        <a
          aria-current={false}
          className="main-nav__item"
          href="/indicators"
          onClick={[Function]}
          style={undefined}
        >
          Indicators
        </a>
      </nav>
    </div>
  </header>
  <div
    className="container"
  >
    <h1>
      content
    </h1>
  </div>
</div>
`;

exports[`Renders error component if error is passed in props 1`] = `
<div
  className="main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <nav
        className="main-nav"
      >
        <a
          aria-current={false}
          className="main-nav__item"
          href="/"
          onClick={[Function]}
          style={undefined}
        >
          Main page
        </a>
        <a
          aria-current={false}
          className="main-nav__item"
          href="/countries"
          onClick={[Function]}
          style={undefined}
        >
          Countries
        </a>
        <a
          aria-current={false}
          className="main-nav__item"
          href="/indicators"
          onClick={[Function]}
          style={undefined}
        >
          Indicators
        </a>
      </nav>
    </div>
  </header>
  <div
    className="container"
  >
    <div
      className="app-error"
    >
      <h1>
        Something went wrong:
      </h1>
      <p>
        Error text
      </p>
    </div>
  </div>
</div>
`;
